stem/output.py

```diff
@@ -0,0 +1,255 @@
+    "CAUCHY_STRESS",
+    "TOTAL_STRESS",
+]
+
+
+def detect_tensor_outputs(requested_outputs: List[str]):
+    """
+    Detects whether gauss point outputs are requested and warns the user
+
+    Args:
+        requested_outputs (List[st]): list of requested outputs (gauss point)
+    """
+    if len(requested_outputs) > 0:
+        detected_tensor_outputs = []
+        for requested_output in requested_outputs:
+            for tensor_output in TENSOR_OUTPUTS:
+                if tensor_output in requested_output:
+                    detected_tensor_outputs.append(tensor_output)
+                    break
+        if len(detected_tensor_outputs):
+            _fmt_list = "".join([f" - {outpt} \n" for outpt in detected_tensor_outputs])
+            _msg = (
+                f"\n[WARNING] tensor output detected:\n{_fmt_list}"
+                f"The outputs are incorrectly rendered for the selected output.\n"
+                f"If vector is selected and ignored if tensor is selected."
+            )
+            print(_msg)
+
+
+@dataclass
+class OutputParametersABC(ABC):
+    """
+    Abstract class for the definition of user output parameters (GiD, VTK, json).
+
+    Attributes:
+        -
+    """
+
+    pass
+
+    @abstractmethod
+    def assemble_parameters(self) -> Dict[str, Any]:
+        """
+        Abstract method for assembling the output properties into a nested dictionary
+        """
+        pass
+
+    @abstractmethod
+    def validate(self):
+        """
+        Abstract method for validating user inputs
+        """
+        pass
+
+    @abstractmethod
+    def is_output_process(self):
+        """
+        Abstract method for checking whether an output is in the output process list
+        (True for GiD, TVK) or not (False for JSON) which is in `processes`
+        """
+        pass
+
+
+@dataclass
+class GiDOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for GiD output
+
+    Attributes:
+        gidpost_flags (Dict[str, Any]):
+        file_label (str):
+        output_control_type (str):
+        output_interval (int):
+        body_output (bool):
+        node_output (bool):
+        skin_output (bool):
+        plane_output (List[str]):
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        point_data_configuration (List[str]):
+    """
+
+    gidpost_flags: Dict[str, Any] = field(default_factory=lambda: DEFAULT_GID_FLAGS)
+    file_label: str = "step"
+    output_control_type: str = "step"
+    output_interval: int = 1
+    body_output: bool = True
+    node_output: bool = False
+    skin_output: bool = False
+    plane_output: List[str] = field(default_factory=lambda: [])
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+    point_data_configuration: List[str] = field(default_factory=lambda: [])
+
+    def assemble_parameters(self) -> Dict[str, Any]:
```

Comment:
i think this function belongs in the Kratos IO. As the only thing it does is preparing the data for kratos